rs/boundary_node/ic_boundary/src/salt_fetcher.rs

```diff
@@ -0,0 +1,152 @@
+use crate::core::Run;
+use anyhow::Error;
+use arc_swap::ArcSwapOption;
+use async_trait::async_trait;
+use candid::Principal;
+use candid::{Decode, Encode};
+use ic_canister_client::Agent;
+use ic_types::CanisterId;
+use prometheus::{
+    register_int_counter_vec_with_registry, register_int_gauge_with_registry, IntCounterVec,
+    IntGauge, Registry,
+};
+use salt_sharing_api::GetSaltResponse;
+use std::time::{SystemTime, UNIX_EPOCH};
+use std::{sync::Arc, time::Duration};
+use tokio::time::sleep;
+use tracing::warn;
+
+fn nonce() -> Vec<u8> {
+    SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_nanos()
+        .to_le_bytes()
+        .to_vec()
+}
+
+struct Metrics {
+    last_successful_fetch: IntGauge,
+    fetches: IntCounterVec,
+    last_salt_id: IntGauge,
+}
+
+impl Metrics {
+    fn new(registry: &Registry) -> Self {
+        Self {
+            last_successful_fetch: register_int_gauge_with_registry!(
+                format!("salt_sharing_last_successful_fetch"),
```

Comment:
I would rename to `anonymization_salt` as the from the perspective of the API BN, it doesn't know that the salt is shared, it just knows that the salt should be used for anonymizing the logs.
```suggestion
                format!("anonymization_salt_last_successful_fetch"),
```